### PR TITLE
Introduce new component EventSourcing

### PR DESCRIPTION
**Pull request by tyx**

### CODE REVIEW

File: src/Component/EventSourcing/.bootstrap.atoum.php [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<?php
```
Comment (shouze) [inline src/Component/EventSourcing/.bootstrap.atoum.php]: This is useless, plz :fire: it!

File: src/Component/EventSourcing/Domain/AggregateHistory.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+namespace Ubirak\Component\EventSourcing\Domain;
+
+final class AggregateHistory extends \SplFixedArray
+{
+    private $aggregateId;
+
+    private $versionId;
+
+    public function __construct(iterable $events, ?int $versionId = null)
```
Comment (shouze) [inline src/Component/EventSourcing/Domain/AggregateHistory.php]: Why not pute int type for versionId and a 0 default value?

File: src/Component/EventSourcing/Domain/Entity.php [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+namespace Ubirak\Component\EventSourcing\Domain;
+
+abstract class Entity extends ChangesRecorder
+{
+    protected $id;
+
+    protected $parentId;
+
+    protected function __construct($id)
```
Comment (shouze) [inline src/Component/EventSourcing/Domain/Entity.php:12]: No type hinting for $id?

File: src/Component/EventSourcing/Domain/AggregateHistory.php (hunk 2 above)
Comment (tyx) [reply src/Component/EventSourcing/Domain/AggregateHistory.php]: yep I will do it

File: src/Component/EventSourcing/Domain/Entity.php (hunk 3 above)
Comment (tyx) [reply src/Component/EventSourcing/Domain/Entity.php:12]: yep, for now Entity accept everything as id

File: src/Component/EventSourcing/composer.json [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    },
+    "autoload-dev": {
+        "psr-4": {
+            "Ubirak\\Component\\EventSourcing\\Tests\\": "Tests"
+        }
+    },
+    "minimum-stability": "dev",
+    "require-dev": {
+        "atoum/atoum": "^3.1"
+    }
```
Comment (shouze) [inline src/Component/EventSourcing/composer.json:27]: what's nice to add is:
```
    "config": {
        "preferred-install": {
            "*": "dist"
        }
    }
```

**Review by shouze — APPROVED** [review]
LGTM

**Pull request merged**